File: packages/health_connector_hc_android/android/src/test/kotlin/com/phamtunglam/health_connector_hc_android/unit_tests/mappers/health_measurement_unit_mappers/TimeDurationMapperTest.kt
package com.phamtunglam.health_connector_hc_android.unit_tests.mappers.health_measurement_unit_mappers

import com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers.toDto
import com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers.toDuration
import com.phamtunglam.health_connector_hc_android.pigeon.TimeDurationDto
import io.kotest.matchers.shouldBe
import kotlin.time.Duration.Companion.hours
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

/**
 * Unit tests for Time Duration Mapper.
 *
 * Tests verify proper bidirectional mapping between [TimeDurationDto] and Kotlin [Duration]
 * objects, including unit conversions between seconds, minutes, and hours.
 */
@DisplayName("TimeDurationMapper")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TimeDurationMapperTest {

    @Test
    @DisplayName(
        "GIVEN TimeDurationDto in seconds → " +
            "WHEN toDuration called → " +
            "THEN creates Duration with correct value",
    )
    fun whenTimeDurationDtoInSeconds_thenCreatesCorrectDuration() {
        // Given
        val dto = TimeDurationDto(seconds = 3600.0)

        // When
        val result = dto.toDuration()

        // Then
        result.toDouble(kotlin.time.DurationUnit.SECONDS) shouldBe 3600.0
    }

    @Test
    @DisplayName(
        "GIVEN Duration object → " +
            "WHEN toDto called → " +
            "THEN converts to DTO in seconds",
    )
    fun whenDurationToDto_thenUsesSeconds() {
        // Given
        val duration = 1.hours

        // When
        val result = duration.toDto()

        // Then
        result.seconds shouldBe 3600.0
    }
}

File: packages/health_connector_hk_ios/test/unit_tests/src/mappers/measurement_unit_mappers/time_duration_mapper_test.dart
import 'package:health_connector_core/health_connector_core_internal.dart';
import 'package:health_connector_hk_ios/src/mappers/measurement_unit_mappers/time_duration_mapper.dart';
import 'package:health_connector_hk_ios/src/pigeon/health_connector_hk_ios_api.g.dart';
import 'package:test/test.dart';

void main() {
  group(
    'TimeDurationMapper',
    () {
      group(
        'TimeDurationToDto',
        () {
          test(
            'converts TimeDuration to TimeDurationDto in seconds',
            () {
              const duration = TimeDuration.seconds(3600.0);
              final dto = duration.toDto();

              expect(dto.seconds, 3600.0);
            },
          );

          test(
            'converts TimeDuration from minutes to seconds',
            () {
              const duration = TimeDuration.minutes(60.0);
              final dto = duration.toDto();

              expect(dto.seconds, 3600.0);
            },
          );
        },
      );

      group(
        'TimeDurationDtoToDomain',
        () {
          test(
            'maps TimeDurationDto to TimeDuration',
            () {
              const value = 3600.0;
              final dto = TimeDurationDto(seconds: value);
              final duration = dto.toDomain();

              expect(duration.inSeconds, value);
            },
          );
        },
      );
    },
  );
}
